client/lib/sites-list/sites-observer.jsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import omit from 'lodash/omit';
+
+/**
+ * Internal dependencies
+ */
+import Debug from 'debug';
+const debug = Debug( 'calypso:site-observe-hoc' );
+
+const sitesObserver = ( WrappedComponent ) => {
+	class SitesObserverComponent extends Component {
+
+		constructor( props, context ) {
+			super( props, context );
+
+			this.update = this.update.bind( this );
+
+			this.state = {};
+		}
+
+		componentWillMount() {
+			this.cacheSites();
+		}
+
+		componentDidMount() {
+			// Using componentDidMount to register to events because
+			// this lifecycle method will not be called on SSR.
+			this.props.sites.on( 'change', this.update );
+		}
+
+		componentWillUnmount() {
+			this.props.sites.off( 'change', this.update );
+		}
+
+		update() {
```

Comment:
might as well use the arrow function syntax here with lexical `this` binding and then we can get rid of the constructor altogether